source/whats-new.txt

```diff
@@ -27,6 +28,38 @@ Learn what's new in:
 * :ref:`Version 3.7 <version-3.7>`
 * :ref:`Version 3.6 <version-3.6>`
 
+.. _version-4.12:
+
+What's New in 4.12
+------------------
+
+New features of the 4.12 {+driver-short+} release include:
+
+- Redefinition of the ``ChangeStream`` class as an async iterable.
+  You can use ``ChangeStream`` instances in any context that expects an
+  ``AsyncIterator``.
+
+  - Notably, change streams can now be used in Javascript ``for-await``
+    loops:
+    
+    .. code-block:: js
+       :copyable: false
+
+       const changeStream = collection.watch();
+       for await (const change of changeStream) {
+         console.log("Received change: ", change);
+       }
+
+- Fix to server monitoring when the driver skips monitoring events. In
+  this release, the driver always updates its view of the topology when
+  processing monitoring events.
+
+- Performance improvements with buffering.
+
+- Modification to data structures used internally in the driver.
```

Comment:
**suggestion (non-blocking) / question:** Would it be more/less effective to merge the last two bullets?  Based on the release notes, looks like the performance improvements are a _result_ of the modified data structures used internally by the drivers, so it might help clarify for users why that last bullet is important.

Can be something along the lines of: 

> Performance improvements to buffering as a result of modified internal data structures used by the driver.